FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice_utest.c

```diff
@@ -627,4 +634,758 @@ void test_task_delay_until_with_config_assert( void )
 
     xTaskDelayUntil( &previousWakeTime , 4 );
     
-}
\ No newline at end of file
+}
+
+/**
+ * @brief xTaskGenericNotify - function to notify a task.
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *              case eNoAction:
+ *              ...
+ *                    break;
+ *
+ * @endcode
+ *
+ *  Cover the eNoAction action type with an explicit task specified.
+ */
+void test_coverage_xTaskGenericNotify_with_eAction_equalto_eNoAction( void )
+{
+    TCB_t xTaskTCBs[ configNUMBER_OF_CORES + 1U ] = { NULL };
+    UBaseType_t xidx = 0;
+    uint32_t *prevValue = NULL;
+    uint32_t ulValue = 0;
+    uint32_t i;
+    BaseType_t xReturn;
+
+    /* Setup the variables and structure. */
+    vListInitialise( &xSuspendedTaskList );
+    vListInitialise( &xPendingReadyList );
+    for( i = 0; i < configNUMBER_OF_CORES; i++ )
+    {
+        xTaskTCBs[ i ].uxPriority = 1;
+        xTaskTCBs[ i ].xTaskRunState = i;
+        xYieldPendings[ i ] = pdFALSE;
+        pxCurrentTCBs[ i ] = &xTaskTCBs[ i ];
+    }
+    /* A suspended task is created to be resumed from ISR. The task has higher priority
+     * than uxTopReadyPriority and the scheduler is suspended. The task will be added
+     * to xPendingReadyList after resumed from ISR. */
+    xTaskTCBs[ configNUMBER_OF_CORES ].uxPriority = 2;
+    listINSERT_END( &xSuspendedTaskList, &xTaskTCBs[ i ].xStateListItem );
+    uxTopReadyPriority = 1;
+    uxSchedulerSuspended = pdTRUE;
+
+    xReturn = xTaskGenericNotify( &xTaskTCBs[0], xidx, ulValue, eNoAction, prevValue);
```

Comment:
Suggest we pass the address of a unit32_t variable and also verify the value updated in this API. The implementation in the commit can also be referenced.
```
uint32_t prevValue;
...

xReturn = xTaskGenericNotify( &xTaskTCBs[0], xidx, ulValue, eNoAction, &prevValue);
```

Reply:
Which test is this, I can add prevValue query and validation. The code quote here doesn't include sufficient context for me to see which test case it is quickly.

Reply:
Ah, a latter test case explicitly tests the pulPreviousNotificationValue branch. I can change this to directly pass NULL rather than use an intermediate pointer to NULL